packages/falso/src/lib/aggregation.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  fake,
+  FakeOptions,
+  getRandomInRange,
+  RandomInRangeOptions,
+} from './core/core';
+
+export interface RandomAggregationOptions
+  extends RandomInRangeOptions,
+    FakeOptions {
+  totalValue?: number;
+}
+
+// https://stackoverflow.com/questions/52489261/typescript-can-i-define-an-n-length-tuple-type
+type Tuple<V, N extends number, T extends V[] = []> = N extends T['length']
+  ? T
+  : Tuple<V, N, [...T, V]>;
+
+/**
+ * Generate a list of numbers that add up to a total value.
+ *
+ * @category general, math
+ *
+ * @example
+ *
+ * randAggregation() // default values is 2 and totalValue is 100
+ *
+ * @example
+ *
+ * randAggregation({ values: 3, totalValue: 1500, length: 10 })
+ *
+ * @example
+ *
+ * randAggregation({ values: 3 })
+ *
+ */
+export function randAggregation<
+  T extends number = 2,
+  Options extends RandomAggregationOptions = never
+>(options?: Options & { values?: T }) {
+  const values: T = (options?.values ?? 2) as unknown as T;
+  const totalValue = options?.totalValue ?? 100;
+
+  if (values <= 1) {
+    throw new Error('Amount must be bigger than 1');
+  }
+
```

Comment:
Why do we need all of that?

```ts
export function randAggregation<
  T extends number = 2,
  Options extends RandomAggregationOptions = never
>(options?: Options & { values?: T }) {
  const values: T = (options?.values ?? 2) as unknown as T;
  const totalValue = options?.totalValue ?? 100;

  if (values <= 1) {
    throw new Error('Amount must be bigger than 1');
  }

  type TupleReturn = number extends T ? number[] : Tuple<number, T>;

  return fake((): TupleReturn => {
    const nums: number[] = [];
    let max = totalValue;

    for (let i = 0; i < values - 1; i++) {
      const num = getRandomInRange({
        min: 0,
        max: getRandomInRange({ min: 0, max })
      });

      max = totalValue - num;
      nums.push(num);
    }

    return [
      ...nums,
      totalValue - nums.reduce((a, b) => a + b, 0),
    ] as unknown as TupleReturn;
  }, options);
}
```

Reply:
I'm not sure I understand what you're asking, sorry. The last part of the function passed into `fake` is shuffling the array.

Reply:
I'm saying that we don't need to shuffle it anymore. Is something wrong with the code I paste?

Reply:
I believe we still need to shuffle it. Since max is decreasing with every iteration of the loop, the odds of a number at the end of the array being large is low

Reply:
ok